### DISCUSSION TITLE
Inquire about cross validation

### DISCUSSION THREAD

**Discussion by Sarath-C-Koppolu** [ultralytics/ultralytics#11725] [General]
Hello,
I am trying to run yolo models. 
my motivation is to run models and see which model is working better with inference time and also see wheather images size has any impact on inference time. so i choose 320,480,640,800,960 as image sizes 

now my question is how do i know that the pretrained model is not resizing to 640? 
for example if i give image size as 320 and use pretrained model which is trained on 640 image size. does the image size is automatically resizing to 640 or it keeps 320 and train the model?

2. I am also using cross validation. i used ultralytics cross validation which at the end i have 5 splits and each split with train and valid and yaml file. 
now if i want to use val with spilt = 'test' so i can get mAP and other metrics for test. how do i apporach for this?

3. my best and last weights are saving at the same time. for instance 90% of the time when i run the model example at 12:00 the weights with best and last are saved automatically at the same time frame even if the last epoch is not ran yet? 
so how do i deal with these.. does this have any effect on training?

**Comment by pderrenger** [comment]
Hello! I'm glad you're experimenting with YOLO models and exploring inference times with different image sizes. Let's address your questions:

1. **Image Resizing**: When you specify an image size (e.g., 320) that differs from the size used in pre-training (e.g., 640), the model will resize the input images to the size you specified (320 in this case). This is done during both training and inference.

2. **Cross Validation and Test Splits**: If you have already established train and validation splits using the provided cross-validation setup and want to evaluate on a 'test' split, you’ll need to modify the respective dataset YAML file. Simply change the validation path to your test set path. For instance:

   ```yaml
   path: ./dataset  # dataset path
   train: images/train  # train images (relative to 'path')
   val: images/test  # test images (replace 'val' with your test set)
   ```

   Then run validation with this configuration to get metrics like mAP on your test data.

3. **Best and Last Weights Saving Simultaneously**: The saving of 'best' and 'last' weights should ideally reflect checkpoints at different stages—'last' being the most recent, and 'best' being those with the highest validation performance. If they appear to save at the same time during incomplete epochs, this could be a session-specific issue or part of a normal behavior depending on interruption or specified save intervals.

I hope this clarifies your concerns. If issues persist, ensure your configurations are consistent and aligned with your objectives. Good luck with your model evaluations! 😊
